pkg/adapter/apiserver/delegate.go

```diff
@@ -44,6 +45,12 @@ func (a *resourceDelegate) Add(obj interface{}) error {
 
 	if result := a.ce.Send(context.Background(), event); !cloudevents.IsACK(result) {
 		a.logger.Errorw("failed to send event", zap.Error(result))
+	} else {
+		eventExt, err := json.Marshal(event.Extensions())
```

Comment:
I would recommend not calling json.Marshal.  This is a non-free method, it would be better to let zap do it if the log line really gets printed by passing the extensions array.